File: web3.py
from web3 import Web3

# Connecting to a local Ethereum node (replace 'http://127.0.0.1:8545' with your node's URL)
w3 = Web3(Web3.HTTPProvider('http://127.0.0.1:8545'))

def get_block_number():
    return w3.eth.block_number

def get_balance(address):
    return w3.eth.get_balance(address)

def get_transaction_count(address):
    return w3.eth.get_transaction_count(address)

def send_transaction(private_key, to_address, value):
    # Example code for sending Ether, handle gas, and other parameters appropriately
    transaction = {
        'to': to_address,
        'value': w3.toWei(value, 'ether'),
        'gas': 2000000,
        'gasPrice': w3.toWei('50', 'gwei'),
        'nonce': w3.eth.getTransactionCount(w3.eth.coinbase),
    }
    signed_tx = w3.eth.account.signTransaction(transaction, private_key)
    tx_hash = w3.eth.sendRawTransaction(signed_tx.rawTransaction)
    return tx_hash.hex()

# Example usage
if __name__ == "__main__":
    block_number = get_block_number()
    print(f"Current block number: {block_number}")

    balance = get_balance('0xYourAddressHere')
    print(f"Balance of address: {balance} wei")

    transaction_count = get_transaction_count('0xYourAddressHere')
    print(f"Transaction count: {transaction_count}")

    # Replace the following with appropriate private key and destination address
    private_key = 'YourPrivateKeyHere'
    to_address = '0xRecipientAddressHere'
    value_to_send = 0.1  # Ether

    tx_hash = send_transaction(private_key, to_address, value_to_send)
    print(f"Transaction sent. Transaction hash: {tx_hash}")

File: web3.js
// web3.js - Example Web3.js Usage

// Import the Web3 library
const Web3 = require('web3');

// Set up an instance of Web3 with an Ethereum node endpoint
const web3 = new Web3('https://mainnet.infura.io/v3/YOUR_INFURA_API_KEY');

// Example usage of Web3.js to interact with Ethereum blockchain

// Get the latest block number
web3.eth.getBlockNumber()
  .then(blockNumber => {
    console.log('Latest block number:', blockNumber);
  })
  .catch(err => {
    console.error('Error:', err);
  });

// Get Ethereum balance for a specific address
const address = '0x1234567890123456789012345678901234567890';
web3.eth.getBalance(address)
  .then(balance => {
    console.log(`Balance of ${address}:`, web3.utils.fromWei(balance, 'ether'), 'ETH');
  })
  .catch(err => {
    console.error('Error:', err);
  });

// Interacting with smart contracts
const contractAddress = '0xContractAddress';
const contractABI = [
  // ... ABI definition of the smart contract
];

const contract = new web3.eth.Contract(contractABI, contractAddress);

// Call a function on the smart contract (example function)
contract.methods.someFunction().call()
  .then(result => {
    console.log('Result of someFunction:', result);
  })
  .catch(err => {
    console.error('Error:', err);
  });

// Sending a transaction to a smart contract (example transaction)
const account = '0xYourAccount';
const privateKey = 'YourPrivateKey';

const functionToCall = contract.methods.someFunction();
const encodedABI = functionToCall.encodeABI();

const txObject = {
  from: account,
  to: contractAddress,
  gas: 2000000,
  data: encodedABI,
};

web3.eth.accounts.signTransaction(txObject, privateKey)
  .then(signedTx => {
    web3.eth.sendSignedTransaction(signedTx.rawTransaction)
      .on('receipt', receipt => {
        console.log('Transaction receipt:', receipt);
      })
      .on('error', err => {
        console.error('Error sending transaction:', err);
      });
  })
  .catch(err => {
    console.error('Error signing transaction:', err);
  });
